int32 size
pointinfo[] points 
